Code/Framework/AzFramework/AzFramework/Quality/QualitySystemComponent.cpp

```diff
@@ -84,6 +91,248 @@ namespace AzFramework
+                {
+                    // object entries (Lua) are not supported yet
+                    AZ_Warning(
+                        "QualitySystemComponent",
+                        false,
+                        "Skipping device attribute rule object entry '%.*s' that is not a string.",
+                        AZ_STRING_ARG(visitArgs.m_fieldName));
+                    return AZ::SettingsRegistryInterface::VisitResponse::Skip;
+                }
+
+                // find the device attribute with this name and evaluate
+                if (auto device = deviceRegistrar->FindDeviceAttribute(visitArgs.m_fieldName); device != nullptr)
+                {
+                    AZ::SettingsRegistryInterface::FixedValueString rule;
+                    if (registry->Get(rule, visitArgs.m_jsonKeyPath))
+                    {
+                        hasMatchingRule = device->Evaluate(rule);
+
+                        // in order for a rule to apply, all device attribute rules must evaluate true
+                        if (!hasMatchingRule)
+                        {
+                            return AZ::SettingsRegistryInterface::VisitResponse::Done;
+                        }
+                    }
+                }
+
+                return AZ::SettingsRegistryInterface::VisitResponse::Continue;
+            };
+
+            AZ::SettingsRegistryVisitorUtils::VisitObject(*registry, callback, jsonPath);
+
+            return hasMatchingRule;
+        }
+
+        bool HasMatchingRule(DeviceAttributeRegistrarInterface* deviceRegistrar, AZ::SettingsRegistryInterface* registry, AZStd::string_view jsonPath)
+        {
+            bool hasMatchingRule = false;
+
+            auto callback = [&](const AZ::SettingsRegistryInterface::VisitArgs& visitArgs)
+            {
+                if (visitArgs.m_type != AZ::SettingsRegistryInterface::Type::Object)
+                {
+                    AZ_Warning(
+                        "QualitySystemComponent",
+                        false,
+                        "Skipping device rules entry '%.*s' that is not an object type.",
+                        AZ_STRING_ARG(visitArgs.m_fieldName));
+                    return AZ::SettingsRegistryInterface::VisitResponse::Skip;
+                }
+
+                if (HasMatchingDeviceAttributeRule(deviceRegistrar, registry, visitArgs.m_jsonKeyPath))
+                {
+                    // only one rule has to match
+                    hasMatchingRule = true;
+                    return AZ::SettingsRegistryInterface::VisitResponse::Done;
+                }
+
+                return AZ::SettingsRegistryInterface::VisitResponse::Continue;
+            };
+
+            AZ::SettingsRegistryVisitorUtils::VisitObject(*registry, callback, jsonPath);
+
+            return hasMatchingRule;
+        }
+
+        struct RuleSetting
+        {
+            AZStd::string setting;
+            AZStd::string value;
+        };

```

Comment:
Can you add a doxygen comment about the RuleSetting struct?

Also can you add `m_` to the struct as well?
```suggestion
        //! Stores the current value of the Rule setting while evaulating
        struct RuleSetting
        {
            AZStd::string m_setting;
            AZStd::string m_value;
        };

```